compiler/rustc_mir/src/transform/check_consts/validation.rs

```diff
@@ -636,12 +638,6 @@ fn place_as_reborrow(
             return None;
         }
 
-        // A borrow of a `static` also looks like `&(*_1)` in the MIR, but `_1` is a `const`
-        // that points to the allocation for the static. Don't treat these as reborrows.
-        if body.local_decls[place.local].is_ref_to_static() {
-            return None;
-        }
-
```

Comment:
This had to be removed in order to keep allowing statics to coerce a `&[T; N]` to a `*[T]`. It is also the piece that causes most of the non-span diagnostic changes.